### ISSUE TITLE
Smart borders seem broken in the presence of emptyWidget

### ISSUE THREAD

**Issue by jtdaugherty** [jtdaugherty/brick#370]
@dmwit - I am wondering if you would be willing to take a look at this issue that I discovered. Essentially, if I put an `emptyWidget` in between two smart borders that I would expect to connect, they don't do so.

Here is a minimal test program. It displays two results, with the first one illustrating the unexpected behavior and the second illustrating the expected behavior:
```
├│

├┤
```

```haskell
{-# LANGUAGE CPP #-}
module Main where

#if !(MIN_VERSION_base(4,11,0))
import Data.Monoid ((<>))
#endif
import Brick
import Brick.Widgets.Border
import Brick.Widgets.Center (center)

ui :: Widget ()
ui = center borderTest

borderTest :: Widget ()
borderTest =
    let a = joinableBorder $ Edges True True False True
        b = joinableBorder $ Edges True True False False
    in joinBorders $
       vBox [ padBottom (Pad 1) $
              a <+> emptyWidget <+>
            , a <+> b
            ]

main :: IO ()
main = simpleMain ui
```

**Comment by dmwit** [comment]
Ahhhh... without having looked at the code yet, I would bet what's going on is that it takes the metadata from along the appropriate "edges" of each widget and stitches them together. So `a <+> emptyWidget` gets the left-hand-side joinableness from `a` and the right-hand-side joinableness from `emptyWidget` (instead of `a`). If that's the case, I think the most obvious solution would be to have a special-case check for zero-width (or zero-height) concatenations.

Let me have a look.

(Is `a <+> emptyWidget <+>` perhaps a mispaste for `a <+> emptyWidget <+> b`? I'm going to start my debugging on the assumption that it is.)

**Comment by jtdaugherty** [comment]
Ah yes, that's right - the 'b' was left out in the paste.

**Comment by jtdaugherty** [comment]
And thank you for looking into this!

**Comment by jtdaugherty** [comment]
This is now fixed in 539a594c2743f2f149bc24e53ba91ed2e7ccd7d3. Thank you, @dmwit!
